components/RegisterISCN.vue

```diff
@@ -0,0 +1,523 @@
+  thumbnailUrl: iscnData.value.coverUrl
+}))
+
+const isFormValid = computed(() => {
+  const requiredFields = {
+    title: !!iscnData.value.title,
+    description: !!iscnData.value.description,
+    authorName: !!iscnData.value.author.name,
+    contentUrl: !!iscnData.value.contentFingerprints.some(f => !!f.url)
+  }
+
+  error.value = Object.entries(requiredFields)
+    .find(([_, isValid]) => !isValid)?.[0]
+    ?.toUpperCase() || ''
+
+  return Object.values(requiredFields).every(Boolean)
+})
+
+onMounted(() => {
+  const initialData = initializeFromSessionStorage()
+  if (initialData) {
+    iscnData.value = initialData
+  }
+  calculateISCNFee()
+})
+
+const initializeFromSessionStorage = () => {
+  if (process.server) {
+    return null
+  }
+
+  try {
+    const storedData = window.sessionStorage.getItem('uploadFileData')
+    if (!storedData) {
+      return null
+    }
+
+    const data: UploadFileData = JSON.parse(storedData)
+
+    const baseData = {
+      type: 'book',
+      title: data.epubMetadata?.title || '',
+      description: stripHtmlTags(data.epubMetadata?.description || ''),
+      isbn: '',
+      publisher: '',
+      publicationDate: '',
+      author: {
+        name: data.epubMetadata?.author || '',
+        description: '',
+        url: ''
+      },
+      license: 'All rights reserved',
+      contentFingerprints: [] as Array<{ url: string }>,
+      downloadableUrls: [] as Array<{
+        url: string;
+        type: string;
+        fileName: string;
+      }>,
+      coverUrl: `ar://${data.epubMetadata?.thumbnailArweaveId}`,
+      language: formatLanguage(data.epubMetadata?.language || ''),
+      tags: data.epubMetadata?.tags || [],
+      ipfsHash: [
+        data.thumbnailIpfsHash,
+        ...data.fileRecords.map(record => record.ipfsHash)
+      ]
+        .filter(Boolean),
+      arweaveId: data.fileRecords
+        .map(record => record.arweaveLink || record.arweaveId)
+        .filter(Boolean)
+    }
+
+    const downloadableFiles = data.fileRecords.filter(
+      record => record.fileType === 'epub' || record.fileType === 'pdf'
+    )
+
+    if (downloadableFiles.length) {
+      baseData.downloadableUrls = downloadableFiles.map(file => ({
+        url: file.arweaveLink || `ar://${file.arweaveId}`,
+        type: file.fileType,
+        fileName: file.fileName
+      }))
+      baseData.contentFingerprints = [
+        ...new Set(
+          data.fileRecords.flatMap(record =>
+            [record.arweaveLink || `ar://${record.arweaveId}`, `ipfs://${record.ipfsHash}`].filter(Boolean)
+          )
+        )
+      ].map(url => ({ url }))
+    }
+
+    if (baseData.contentFingerprints.length === 0) {
+      baseData.contentFingerprints = [{ url: '' }]
+    }
+    if (baseData.downloadableUrls.length === 0) {
+      baseData.downloadableUrls = [{ url: '', type: '', fileName: '' }]
+    }
+
+    return baseData
+  } catch (error) {
+    console.error('Error reading from sessionStorage:', error)
```

Comment:
When is the `sessionStorage` cleared?

Reply:
I plan to clear sessionStorage right after successfully minting NFT in the next PR